### PR TITLE
Process api hook

### PR DESCRIPTION
**Pull request by matthid**
### Description

Use the new API in order to write unit-tests

### CODE REVIEW

File: src/app/Fake.DotNet.Testing.NUnit/NUnit3.fs [hunk 1]
```diff
@@ -309,35 +309,50 @@ let buildArgs (parameters:NUnit3Params) (assemblies: string seq) =
...
-    __.MarkSuccess()
+    let generatedArgs = buildArgs parameters assemblies
+    //let processTimeout = TimeSpan.MaxValue // Don't set a process timeout. The timeout is per test.
+    
+    let path = Path.GetTempFileName()
+    let args = (sprintf "@%s" path)
+    CreateProcess.fromRawWindowsCommandLine tool args
+    |> CreateProcess.withFramework
+    |> CreateProcess.withWorkingDirectory (getWorkingDir parameters)
+    //|> CreateProcess.withTimeout processTimeout
```
Comment (BlythMeister) [inline src/app/Fake.DotNet.Testing.NUnit/NUnit3.fs]: I think we want this max timeout to ensure that the process doesn;t get shutdown.
Nunit has it's own timeout built in as an argument.

File: src/app/Fake.DotNet.Testing.NUnit/NUnit3.fs [hunk 2]
```diff
@@ -309,35 +309,50 @@ let buildArgs (parameters:NUnit3Params) (assemblies: string seq) =
...
+            match result with
+            | OK -> ()
+            | _ -> raise (FailedTestsException(errorDescription result))
+    )
+
+let run (setParams : NUnit3Params -> NUnit3Params) (assemblies : string seq) =
+    let assemblies = assemblies |> Seq.toArray
+    let details = assemblies |> String.separated ", "
+    use __ = Trace.traceTask "NUnit" details
+    let p = createProcess setParams assemblies
```
Comment (BlythMeister) [inline src/app/Fake.DotNet.Testing.NUnit/NUnit3.fs]: could this not be
```
createProcess setParams assemblies
|> Proc.run
```

File: src/app/Fake.DotNet.Testing.NUnit/NUnit3.fs (hunk 1 above)
Comment (matthid) [reply src/app/Fake.DotNet.Testing.NUnit/NUnit3.fs]: The `CreateProcess` has no default timeout, therefore it waits forever by default which seems like the intended behavior.

File: src/app/Fake.DotNet.Testing.NUnit/NUnit3.fs (hunk 2 above)
Comment (matthid) [reply src/app/Fake.DotNet.Testing.NUnit/NUnit3.fs]: yes indeed

File: src/app/Fake.DotNet.Testing.NUnit/NUnit3.fs (hunk 1 above)
Comment (BlythMeister) [reply src/app/Fake.DotNet.Testing.NUnit/NUnit3.fs]: Fair enough!

File: src/test/Fake.Core.UnitTests/Fake.Core.Process.fs [hunk 3]
```diff
@@ -11,14 +11,15 @@ let fsCheckConfig = { FsCheckConfig.defaultConfig with maxTest = 1000  }
 [<Tests>]
 let tests = 
   testList "Fake.Core.Process.Tests" [
-    testPropertyWithConfig fsCheckConfig "toWindowsCommandLine is the inverse of fromWindowsCommandLine" <|
+    //Process.setEnableProcessTracing true
```
Comment (matthid) [inline src/test/Fake.Core.UnitTests/Fake.Core.Process.fs]: If this line is uncommented it completely breaks the "fake-context" for some reason and about the half (around 30) tests start to fail... Would be interesting to figure out why.
Note that not only tests in this file start failing but everywhere (especially the `FakeVar` and `Context` ones)

**Pull request merged**